WordPress/gutenberg
# Storybook: Upgrade to v8.0.x

## What?
Upgrading Storybook to v8.0. 

## Why?
We've been stuck on Storybook 7.x for a while.

## How?
Doing a gradual migration, one step at a time, migrating all necessary changes for 8.0.

Planning to incrementally get it to the latest version in subsequent PRs. 

Use my self-review for more details on each change. 

## Testing Instructions
* Verify storybook builds
* Test every component or docs page to ensure nothing breaks.

### Testing Instructions for Keyboard
Same

## Screenshots or screencast 

None
